pytorch_lightning/callbacks/rich_model_summary.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Union
+
+from pytorch_lightning.callbacks import ModelSummary
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _RICH_AVAILABLE
+from pytorch_lightning.utilities.model_summary import get_human_readable_count
+
+if _RICH_AVAILABLE:
+    from rich.console import Console
+    from rich.table import Table
+
+
+class RichModelSummary(ModelSummary):
+    def __init__(self, max_depth: int = 1) -> None:
+        if not _RICH_AVAILABLE:
+            raise MisconfigurationException(
+                "`RichModelSummary` requires `rich` to be installed. Install it by running `pip install rich`."
+            )
+        super().__init__(max_depth)
+
+    @staticmethod
+    def summarize(
+        summary_data: List[List[Union[str, List[str]]]],
+        total_parameters: int,
+        trainable_parameters: int,
+        model_size: float,
+    ) -> None:
+
+        console = Console()
+
+        table = Table(title="Model Summary")
+
+        table.add_column(" ")
+        table.add_column("Name", justify="left", style="cyan", no_wrap=True)
+        table.add_column("Type", style="magenta")
```

Comment:
Regarding our offline conversation, how could you see these styles being easier to modify? I was thinking some kinda config dict structure that the user can pass, or we just expose the colours to the `__init__`

Reply:
Yup, after your recent RichProgressBar PR gets merged, will add a config dataclass argument here to the signature as well!